src/d3d12/d3d12_texture_resources.hpp

```diff
@@ -22,12 +22,18 @@ namespace wr::d3d12
 		ID3D12Resource* m_resource;
 		ID3D12Resource* m_intermediate;
 		ResourceState m_current_state;
-		DescriptorAllocation m_desc_allocation;
+		DescriptorAllocation m_srv_allocation;
+		DescriptorAllocation m_uav_allocation;
+
+		//This allocation can be either 1 or 6 continous descriptors based on m_is_cubemap
+		std::optional<DescriptorAllocation> m_rtv_allocation = std::nullopt;
 
 		uint8_t* m_allocated_memory;
+		size_t m_needed_memory;
```

Comment:
please use std::size_t. size_t are C headers and if we all use std:: everywhere its more consistent.